### PR TITLE
Introduce queue abstraction

### PR DESCRIPTION
**Pull request by kkaneda**
No functional change, but introduce a queue when processing chat completion requests.

This is for up-coming change to support multi-cluster environment.

### CODE REVIEW

File: server/internal/infprocessor/infprocessor.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	v1 "github.com/llm-operator/inference-manager/api/v1"
+	"github.com/llm-operator/inference-manager/common/pkg/models"
+)
+
+const (
+	taskQueueSize  = 100
+	completionPath = "/v1/chat/completions"
+)
+
+// Task is an infeference task.
```
Comment (guangrui-cloudnatix) [inline server/internal/infprocessor/infprocessor.go:21]: "inference"

File: server/internal/infprocessor/infprocessor.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+type Task struct {
```
Comment (guangrui-cloudnatix) [inline server/internal/infprocessor/infprocessor.go:22]: not sure, wondering if we need to preserve  `context` also?

File: server/internal/infprocessor/infprocessor.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+// Run runs the processor.
+func (p *P) Run(ctx context.Context) error {
+	for {
+		t, err := p.queue.Dequeue(ctx)
+		if err != nil {
+			return err
+		}
+		p.runTask(ctx, t)
```
Comment (guangrui-cloudnatix) [inline server/internal/infprocessor/infprocessor.go:92]: bypass `runTask` if t is nil?

Comment (guangrui-cloudnatix) [reply server/internal/infprocessor/infprocessor.go:92]: also, it seems a busy-loop, not sure if it should sleep somewhere?

**Review by guangrui-cloudnatix — APPROVED** [review]

File: server/internal/infprocessor/infprocessor.go (hunk 1 above)
Comment (kkaneda) [reply server/internal/infprocessor/infprocessor.go:21]:
```suggestion
// Task is an inference task.
```

File: server/internal/infprocessor/infprocessor.go (hunk 3 above)
Comment (kkaneda) [reply server/internal/infprocessor/infprocessor.go:92]: Ah `t` should be non-nil when `err` is `nil`. Also Dequeue blocks when there is no task as the channel read is blocking by default.

File: server/internal/infprocessor/infprocessor.go (hunk 2 above)
Comment (kkaneda) [reply server/internal/infprocessor/infprocessor.go:22]: good point! not sure. let me add a TODO.
```suggestion
// TODO(kenji): Consider preserving the request context as well.
type Task struct {
```

**Pull request merged**